src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -311,13 +314,31 @@ private void generateKeys(List<BibEntry> entries) {
     public List<BibEntry> handleBibTeXData(String entries) {
         BibtexParser parser = new BibtexParser(preferencesService.getImportFormatPreferences(), fileUpdateMonitor);
         try {
-            return parser.parseEntries(new ByteArrayInputStream(entries.getBytes(StandardCharsets.UTF_8)));
+            List<BibEntry> result = parser.parseEntries(new ByteArrayInputStream(entries.getBytes(StandardCharsets.UTF_8)));
+            List<BibtexString> stringConstants = parser.getStringValues();
+            importStringConstantsWithDuplicateCheck(stringConstants);
+            return result;
         } catch (ParseException ex) {
             LOGGER.error("Could not paste", ex);
             return Collections.emptyList();
         }
     }
 
+    public void importStringConstantsWithDuplicateCheck(List<BibtexString> stringConstants) {
+        for (BibtexString stringConstantToAdd : stringConstants) {
+            try {
+                ConstantsItemModel checker = new ConstantsItemModel(stringConstantToAdd.getName(), stringConstantToAdd.getContent());
+                if (checker.combinedValidationValidProperty().get()) {
+                    bibDatabaseContext.getDatabase().addString(stringConstantToAdd);
+                } else {
+                    dialogService.showErrorDialogAndWait(Localization.lang("Pasted string constant \"%0\" was not added because it is not a valid string constant", stringConstantToAdd.getName()));
```

Comment:
I have a better idea:
1. Make a list with strings constants that were imported/not imported
2. Display them at the end of the loop, so the user does not have to click yes/no etc for each string contant